test/common/router/config_impl_test.cc

```diff
@@ -2029,5 +2034,229 @@ TEST(RoutePropertyTest, excludeVHRateLimits) {
   EXPECT_TRUE(config_ptr->route(headers, 0)->routeEntry()->includeVirtualHostRateLimits());
 }
 
+TEST(CustomRequestHeadersTest, AddNewHeader) {
```

Comment:
the two test cases added are inline with other tests in config_impl_test.cc where we have a simple virtual hosts config and we call the route logic with mock headers.  Do you just want me to create a separate test file like req_header_formatter_test.cc and move these two test cases in there ?